moveit_ros/moveit_servo/test/servo_launch_test_common.hpp

```diff
@@ -75,26 +74,38 @@ class ServoFixture : public ::testing::Test
 public:
   void SetUp() override
   {
+    ASSERT_TRUE(servo_parameters_loaded_success_);
     executor_->add_node(node_);
     executor_thread_ = std::thread([this]() { this->executor_->spin(); });
   }
 
   ServoFixture()
     : node_(std::make_shared<rclcpp::Node>("servo_testing"))
     , executor_(std::make_shared<rclcpp::executors::SingleThreadedExecutor>())
-    , parameters_(getTestParameters())
-    , PUBLISH_HZ(2.0 / parameters_->incoming_command_timeout)
-    , PUBLISH_PERIOD(1.0 / PUBLISH_HZ)
-    , TIMEOUT_ITERATIONS(10 * PUBLISH_HZ)
-    , publish_loop_rate_(PUBLISH_HZ)
-    , servo_server_node_name_("/servo_server")
   {
+    // read parameters and store them in shared pointer to constant
+    {
+      auto servo_parameters = std::make_shared<moveit_servo::ServoParameters>();
+      servo_parameters_loaded_success_ = moveit_servo::readParameters(servo_parameters, node_, LOGGER);
+      servo_parameters_ = servo_parameters;
+    }
```

Comment:
@henningkayser here is a way to load parameters into const storage.  This uses the shared_ptr to a const as a class variable: `servo_parameters_` and a local non-const one.  The parameters are loaded into the temporary non-const parameters and then the shared pointer is given the ownership and the local non-const goes out of scope.

Reply:
Nice solution! I had something similar in mind (more inside the struct for const/non-const parameters) but this is probably the way to go for the beginning.

Reply:
Can you add a further comment at L86.5 about why that matters? For those of us who are new to ROS2   ;)

Maybe something like:

`The constant can be shared by many individual node components`